front/pages/w/[wId]/members/index.tsx

```diff
@@ -566,9 +604,21 @@ function ChangeMemberModal({
     <Modal
       isOpen={showModal}
       onClose={onClose}
-      hasChanged={false}
+      isSaving={isSaving}
+      hasChanged={
+        selectedRole !== null && selectedRole !== member.workspaces[0].role
+      }
       title={member.name || "Unreachable"}
       type="right-side"
+      onSave={async () => {
+        setIsSaving(true);
+        // we know selectRole is not null because of the hasChanged check
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        await handleMemberRoleChange(member, selectedRole!);
```

Comment:
We generally have our functions defined above the JSX. Can we move handleMemberRoleChange above?